### PR TITLE
Implement new `key-store`

### PR DESCRIPTION
**Pull request by nduchak**
https://www.pivotaltracker.com/story/show/161319079
Modify keystore to use `xsalsa20-poly1305` encryption and `argon2id` as derived key function.

### CODE REVIEW

**Comment by ricricucit** [comment]
This PR is going to wait before merge.
The main problem is that the specification given to us _might_ be wrong.

So @nduchak is checking if [this](https://github.com/aeternity/hd-wallet-js/blob/master/src/hd-key.js) is not conflicting with what he implemented.

So now there are 2 possible paths:
1. update the PR, for which we would need new specifications and we'll implement something new again :facepalm: 
2. [this](https://github.com/aeternity/hd-wallet-js/blob/master/src/hd-key.js) is "wrong" and we followed the right specifications, so hd-walled needs to change (and I'll accept the PR)

I should know this by today because, @emil-apeunit is investigating this issue.

**Comment by ricricucit** [comment]
**Update/Blocker**
ATM we still have 2 problems or better "inconsistencies" that we should try to standardise.
They are related to how the [HD-Walled-JS](https://github.com/aeternity/hd-wallet-js) and the SDKs currently give names and work with keys:

1. **Naming**: We call the keys `pub/priv` for the SDKs and `privateKey/publicKey` for [HD-wallet-JS accounts](https://github.com/aeternity/hd-wallet-js/blob/master/test/hd-wallet.js). We should find a naming compromise (standardisation). My suggestion is to use `secretKey/publicKey` in the SDKs (since they are already being used by some aepps)
2. **different bytes length**? 32 bytes for chainCode (HDWallet) and 64 bytes for priv (SDK). What is the right length?

[This](https://github.com/aeternity/aepp-sdk-js/pull/41#issue-191069317) and [this](https://github.com/aeternity/hd-wallet-js/blob/master/test/hd-wallet.js) explains something more on how the hd-wallet-js is used...so I can't understand if this is an actual issue.

@davidyuk can you help me?

**Comment by davidyuk** [comment]
@ricricucit sure!

1. `publicKey`/`secretKey` in `hd-wallet-js` is proxied from `tweetnacl`, I did not find a reason to change it.
2. `priv` key in SDK has the same length as `secretKey` from `hd-wallet-js` (64 bytes), because of `tweetnacl`. For some reason `nacl.sign.keyPair.fromSeed` concatenates the private key (actually it is a passed seed) with the public key and puts in into `secretKey` field. Chain code can't be used as a private key, it is only for derivation process.

**Comment by ricricucit** [comment]
Thanks! We will change our `priv/pub` with `publicKey/secretKey` for consistency across systems.

File: test/cli/index.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ import { generateKeyPair } from '../../es/utils/crypto'
 
 const cliCommand = './bin/aecli.js'
 
-const url = process.env.TEST_URL || 'http://localhost:3013'
+const url = process.env.TEST_URL || ''
```
[the remark this answers is not in the document]
Comment (nduchak) [reply test/cli/index.js]: Sorry. Will revert it

**Pull request merged**